nativelib/src/main/resources/dirent.c

```diff
@@ -1,19 +1,29 @@
+#ifndef _WIN32
 #include <dirent.h>
+#else
+#include "os_win_dirent.h"
```

Comment:
While mimicking POSIX seems to still work for the small scale of the API we're using now I'm really not sure if it's going to scale longer term once threads, sockets etc are added to the mix. We're not going to reimplement all of the POSIX, this is just too much work. 

A much more reasonable option would be to expose Windows-specific APIs on Windows and use those to implement corresponding parts of the java library.